src/OrbitGl/TimelineUi.cpp

```diff
@@ -73,6 +76,12 @@ void TimelineUi::RenderLabel(PrimitiveAssembler& primitive_assembler, TextRender
 
   std::string label = GetLabel(tick_ns, number_of_decimal_places);
   float world_x = GetTickWorldXPos(tick_ns);
+  // Checking that at least some part of the label will be visible.
+  if (world_x + text_renderer.GetStringWidth(label.c_str(), layout_->GetFontSize()) < GetPos()[0] ||
+      world_x > GetPos()[0] + GetSize()[0]) {
+    return;
+  }
```

Comment:
This is close to being too complex for my brain to easily understand. I think splitting the condition up like this helps:
```
bool label_is_left_of_timeline = world_x + text_renderer.GetStringWidth(label.c_str(), layout_->GetFontSize()) < GetPos()[0];
bool label_is_right_of_timeline =  world_x > GetPos()[0] + GetSize()[0];
if (label_is_left_of_timeline || label_is_right_of_timeline) return;
```
I hope I understood what is going on here

Reply:
Yes, you got it 100%. I kept thinking and I decided to use intervals intersections, what do you think?